sympy/geometry/point.py

```diff
@@ -404,54 +371,55 @@ def ambient_dimension(self):
         """The dimension of the ambient space the point is in.
         I.e., if the point is in R^n, the ambient dimension
         will be n"""
-        return len(self)
+        return getattr(self, '_ambient_dimension', len(self))
 
     @property
     def orthogonal_direction(self):
-        """Returns a non-zero point in the x-y plane that is
-        orthogonal to the line containing `self` and the origin.
+        """Returns a non-zero point that is orthogonal to the
+        line containing `self` and the origin.
 
         Examples
         ========
 
         >>> from sympy.geometry import Line, Point
         >>> a = Point(1, 2, 3)
         >>> a.orthogonal_direction
-        Point2D(-2, 1)
+        Point3D(-2, 1, 0)
         >>> b = _
         >>> Line(b, b.origin).is_perpendicular(Line(a, a.origin))
         True
         """
+        dim = self.ambient_dimension
         # if a coordinate is zero, we can put a 1 there and zeros elsewhere
         if self[0] == S.Zero:
-            return Point2D(1, 0)
+            return Point([1] + (dim - 1)*[0])
         if self[1] == S.Zero:
-            return Point2D(0, 1)
-        # if the first two coordinates aren't zero, we can create a
-        # non-zero orthogonal vector by swapping them and negating
-        # one (x is arbitrarily chosen)
-        return Point2D(-self[1], self[0])
+            return Point([0,1] + (dim - 2)*[0])
+        # if the first two coordinates aren't zero, we can create a non-zero
+        # orthogonal vector by swapping them, negating one, and padding with zeros
+        return Point([-self[1], self[0]] + (dim - 2)*[0])
 
     @property
     def unit(self):
         """Return the Point that is in the same direction as `self`
         and a distance of 1 from the origin"""
-        return self/abs(self)
+        return self / abs(self)
 
     @staticmethod
     def affine_rank(*points):
         """The affine rank of a set of points is the dimension
         of the smallest affine space containing all the points.
         For example, if the points lie on a line (and are not all
         the same) their affine rank is 1.  If the points lie on a plane
-        but not a line, their affine rank is 2."""
+        but not a line, their affine rank is 2.  By convention, the empty
+        set has affine rank -1."""
 
         if len(points) == 0:
-            raise TypeError("Must have at least one point")
+            return -1
 
         # make sure we're genuinely points
         # and translate every point to the origin
-        points = Point.normalize_dimensions(*[Point(i) for i in points])
```

Comment:
If you don't normalize you can get a Matrix error with different length rows: `Point.affine_rank(Point(1,2),Point(1,2,3))`